Source/Examples/Example01_windows_blit_tree/Main.cpp

```diff
@@ -0,0 +1,205 @@
+FrameBuffer<PixelFormat::BGRA> frameBuffer(nullptr, imageWidth, imageHeight);
+
+static XYZ pixelColor(255, 0, 0);
+static uint8_t pixelAlpha = 255;
+static int pixelPosX = 400;
+static int pixelPosY = 400;
+
+static XYZ truckColor(165, 42, 0); //�����֦����ɫ
+static XYZ leafColor(0, 255, 0); //��Ҷ����ɫ
+constexpr int truckWidth = 28; //��ɵĿ��
+constexpr int truckHeight = 200; //��ɵĸ߶�
+constexpr int truckPosX = imageWidth/2;  //���������Ĵ������X(��1����辿��洦)
+constexpr int truckPosY = imageHeight-truckHeight;  //���������Ĵ������Y(��1����辿��洦)
+constexpr float branchWidthRatio = 0.55f;  //n+1�����֦��n�����֦�ֶȵı�ֵ����֦Խ����Խϸ
+constexpr float branchHeightRatio = 0.75f; //n+1�����֦��n�����֦���ȵı�ֵ����֦Խ����Խ��
+constexpr int branchDegree = 19; //n+1�����֦��n�����֦�ļн�
+constexpr int HighestLevel = 7; //������ܳɳ����ٲ�
+
+bool isAnimationPlay = true; //�Ƿ񲥷���Ķ�����true�����Ŷ�����false��ֱ����ɽ��
+constexpr float aTimeInterval = 0.5f; //���Ŷ���ʱ��ÿ���һ����֦��ʱ����
+Timer aTimer;
+
+int main()
+{
+    Color clearScreenColor(0, 0, 0);
+
+    Application application(imageWidth, imageHeight, L"SoftRenderer");
+    application.init();
+    frameBuffer.setData(application.getFrameBuffer());
+    frameBuffer.clear(clearScreenColor);
+
+    Timer timer;
+    timer.init();
+
+    std::thread renderThread = std::thread(drawTree);
+    renderThread.detach();
+
+    constexpr float frameTime = 1.0f / 60.0f;
+    float totalTime = 0.0f;
+    while (true)
+    {
+        // Timer
+        float deltaTime = timer.tick();
+        if (totalTime < frameTime)
+        {
+            totalTime += deltaTime;
+            continue;
+        }
+        else
+        {
+            totalTime = 0.0f;
+        }
+
+        // Present the frame buffer
+        application.present();
+
+        // Message
+        application.processMessages();
+    }
+    return 0;
+}
+
+//�ݹ������
+void drawTree()
+{
+    //�������
+    for (int x = -truckWidth/2; x < truckWidth/2; ++x)
+    {
+        for (int y = 0; y < truckHeight; ++y)
+        {
+            frameBuffer.fill(truckPosX+x, truckPosY+y, truckColor, pixelAlpha);
+        }
+    }
+    
+    //������辺���Ҷ
+    int l_treeLevel = 0; //��Ĳ��������0�㣬ÿ��һ�β�������һ��
+    int l_rootX = truckPosX; //ÿһ����辵ĸ�ڵ�
+    int l_rootY = truckPosY; //ÿһ����辵ĸ�ڵ�
+    int l_branchWidth = (int)(truckWidth * branchWidthRatio); //��辵Ĵֶȣ�Խ���ϵ����Խϸ
+    int l_branchHeight = (int)(truckHeight * branchHeightRatio); //��辵ĳ��ȣ�Խ���ϵ����Խ��
+    int l_branchDegree = branchDegree; //ÿһ����辷ֲ�ĽǶ�
+
+    aTimer.init();
+
+    drawBranch(l_treeLevel, l_rootX, l_rootY, l_branchWidth, l_branchHeight, l_branchDegree);
+    drawBranch(l_treeLevel, l_rootX, l_rootY, l_branchWidth, l_branchHeight, -l_branchDegree);
+}
+
+void drawBranch(int treeLevel, int rootX, int rootY, int branchWidth, int branchHeight, int degree)
+{
+    treeLevel++;
+    if (treeLevel > HighestLevel) return;
+
+    if (isAnimationPlay==true)
```

Comment:
```suggestion
    if constexpr (isAnimationPlay)
```
C++ 17 feature `if constexpr` can reduce if branch in compile time. It is nice to use it widely.